includes/abilities-api/class-wp-ability.php

```diff
@@ -334,6 +348,10 @@ protected function do_execute( array $input ) {
 			);
 		}
 
+		if ( empty( $this->get_input_schema() ) && null === $input ) {
```

Comment:
From a defensive coding perspective, we should not let any input go through that hasn't been validated. If there's no input schema, that's always the case.

```suggestion
		if ( empty( $this->get_input_schema() ) ) {
```

In practice, this change does not make any difference. But it helps prevent accidental problems down the line, e.g. if someone later changes `validate_input()` to behave differently.

Reply:
Good catch. It’s an artifact from my initial implementation.